src/lib/y2storage/dialogs/callbacks/activate_luks.rb

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+Yast.import "UI"
+
+module Y2Storage
+  module Dialogs
+    module Callbacks
+      # Dialog to manage luks activation callbacks
+      class ActivateLuks
+        include Yast::UIShortcuts
+        include Yast::I18n
+
+        attr_reader :encryption_password
+
+        def initialize(uuid, attempt)
+          @uuid = uuid
+          @attempt = attempt
+        end
+
+        def run
+          create_dialog
+
+          result = loop do
+            input = Yast::UI.UserInput
+            case input
+            when :password
+              activate_button
+            when :accept
+              @encryption_password = password
+              break input
+            when :cancel, :abort
+              break input
+            end
+          end
+
+          Yast::UI.CloseDialog
+          result
+        end
+
+      protected
+
+        attr_reader :uuid, :attempt
+
+        def create_dialog
+          Yast::UI.OpenDialog(dialog_content)
+          activate_button
+        end
+
+        # rubocop:disable  Metrics/MethodLength
+        def dialog_content
+          VBox(
+            VSpacing(0.4),
+            HBox(
+              HSpacing(1),
+              VBox(
+                Left(Heading(_("Encrypted Volume Activation"))),
+                VSpacing(0.2),
+                Left(
+                  Label(
+                    _("The following device contain an encryption signature but the \n" \
+                        "password is not yet known.")
+                  )
+                ),
+                VSpacing(0.2),
+                Left(Label("UUID: #{uuid}")),
+                VSpacing(0.2),
+                Left(
+                  Label(
+                    _("The password need to be known if the device is needed either \n" \
```

Comment:
Yes, it is a bit strange. Since it is a little different to the original one, which was refereed to several devices, I guess we could rewrite it.